mkosi/__init__.py

```diff
@@ -1521,103 +1531,79 @@ def mount_overlay(
                f'upperdir={realroot}',
                f'workdir={workdir.name}']
 
-    do_mount("overlay", realroot, options, type="overlay", read_only=read_only)
-    mount_bind(realroot, root)
-    return realroot, workdir
-
-
-@complete_step("Cleaning up overlayfs")
-def clean_up_overlay(root: Path, realroot: Path, workdir: TempDir) -> None:
-    """Destroy the overlayfs structure set up by `mount_overlay`.
-
-    When this function returns, the contents of the root file system have been
-    moved into root, and `realroot` and `workdir` are gone.
-
-    If `realroot` is set, it means we mounted `root` twice: the first mount is
-    the overlayfs mount, and the second is a bind-mount to adjust the location
-    one level up. Thus we need unmount twice too; after the first unmount here,
-    the image remains mounted at `root`.
-    """
-
-    umount(root)
-    umount(realroot)
-
-    workdir.cleanup()
-
-    # Let's now move the contents of realroot into root
-    for entry in os.scandir(realroot):
-        os.rename(realroot / entry.name, root / entry.name)
-    realroot.rmdir()
+    try:
+        with workdir, mount("overlay", realroot, options=options, type="overlay", read_only=read_only), mount_bind(realroot, root):
```

Comment:
I think it's one of the cases where judicious use of line continuations would be more readable…

Reply:
I moved the call to `mount` out of the `with` statement into a variable to shorten the line